frontend/views/GameDetailedView.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { AppStore } from '../storage/common';
+import { Game } from '../storage/models/Game';
+import { GameSeries } from '../storage/models/GameSeries';
+import { Image, undefinedCover } from '../storage/models/Image';
+import { Narrow } from '../components/styles/Narrow';
+import { PageHeader } from '../components/styles/PageHeader';
+import { PageTitle } from '../components/styles/PageTitle';
+import { PageSubtitle } from '../components/styles/PageSubtitle';
+import { PageSection } from '../components/styles/PageSection';
+import { SectionHeader } from '../components/styles/SectionHeader';
+import { GameBox } from '../components/GameBox';
+import { RouteComponentProps, Redirect } from 'react-router-dom';
+
+interface MatchParams {
+	gameId: string;
+}
+interface GameDetailevViewProps {
+	games: Map<number, Game>;
+	gameSeries: Map<number, GameSeries>;
+	images: Map<number, Image>;
+	chosenGameId: number;
+}
+interface GameDetailedViewState {
+	chosenGame: Game;
+}
+
+export class GameDetailedView extends React.Component<GameDetailevViewProps, GameDetailedViewState> {
+	gameFound: boolean;
+	chosenSeries: GameSeries;
+	chosenImage: Image;
+	constructor(props: GameDetailevViewProps) {
+		super(props);
+		if (props.games.has(props.chosenGameId)) {
+			this.gameFound = true;
+			this.state = {
+				chosenGame: props.games.get(props.chosenGameId)!,
+			};
+			this.chosenSeries = props.gameSeries.get(this.state.chosenGame.seriesId)!;
+			if (!this.chosenSeries) {
+				throw new ReferenceError(
+					`Game with id: ${props.chosenGameId} have a missing game series associated with it. Game Series Id:${this.state.chosenGame.seriesId}`
+				);
+			}
+			this.chosenImage = props.images.get(this.state.chosenGame.imageId) || undefinedCover;
+		} else {
+			this.gameFound = false;
```

Comment:
There must be a better way to do this but leave it as-is for now.

Reply:
Okay, I don't see a better way. This is crap and I hate this, so here's a proposition:

 1. For each view have a separate `<name>Route` component, whose only job is to check preconditions and render the actual view.
 2. This would mean move the store logic to the `Route` components, and make the `View` components get the final data as props.
 3. Thus Views will become simpler, and will always work from the assumption that the data they have is valid.

It's probably best to do this after this PR is merged, so keep that in mind as your next step.